### DISCUSSION TITLE
Feedback: Small Issues with Nebula App

### DISCUSSION THREAD

**Discussion by canadaduane** [creativelifeform/three-nebula#151] [General]
Hey, I love what you're doing! Keep up the great work.

Here are some observations while using the Nebula app for the first time:

- download was quite large (360MB) which was hard to get while at a cabin in the woods :)
- The first thing I tried to do was rotate a particle by adding a "Rotate" particle behavior. I couldn't get the `twirl_01` texture to do anything rotationally.
- I wasn't sure what units the rotation values are in ("1 degree" or "1 radian"?)
- I tried editing the "Spring" value (0.01) in the "Spring" section of Particle behaviors, but the number seems to reset itself to 0.01
- when trying to "backspace" a numeric value, it won't let me delete the last character (presumably "" is not a valid number, or something). As a workaround, I used "shift+left arrow key" to highlight the number, and this seems to trigger a separate keyboard shortcut that causes the camera/canvas to move a few pixels to the left. (But it still let me highlight & then change the number, so yay!)

I'm looking forward to integrating this into our virtual world, Relm (https://www.relm.us).

BTW, the minified 100kb filesize is somewhat large for us, since we want website load times to be quick. However, there is a lot of functionality packed in here. Just wanted to voice the value so it can be considered as the library's features change or grow.

Thanks again, and nice work!

**Comment by rohan-deshpande** [comment]
Hi thanks for using the app!

Apart from the app size almost all of these bugs have been fixed. I'll be cutting a new release soon.

In terms of rotating, this is going to require a bit more work especially for the GPURenderer. Doing it with the SpriteRenderer might be easy, but I don't want to concentrate on this renderer anymore as the performance is so poor. 

App size, hmm, there's not much that can be done about that. I can take a look at my dependencies but there is a fair amount of stuff going on, it's a relatively complex app and it has to include all of Chromium along with it. 

Would really be interested in seeing what you are doing with it! I'm working on a gallery site soon called Nebula Voyager which I hope can build a bit of a community around the app and what it can do.

**Comment by canadaduane** [comment]
Oh, interesting, I didn't realize the SpriteRenderer isn't using the GPU.
Is there a write-up somewhere that describes the pros and cons of each
renderer and the constraints under which they will work?

I understand about the size of the app.  Electron is a wonderful
cross-platform gift with a heavy cost in bytes shipped. Better to have
something like this app across all major desktop OSes than nothing!
